### DISCUSSION TITLE
Add feature from rct classic REPAIR benches lamps and trashbins

### DISCUSSION THREAD

**Discussion by 733737** [OpenRCT2/OpenRCT2#17565] [Feature requests]
in rct classic scenery menu you can enter a repair mode where all broken path items are highlighted red and you can click to repair them

**Comment by ocalhoun6** [comment]
Well, we already have a cheat that instantly repairs all vandalism ... but maybe a non-cheating version would be nice to have. One that still charges you money.

We also already have a 'view path issues' mode that highlights any vandalized scenery. Maybe that could come with an additional option to easily repair vandalism?

One option would be to have handymen or mechanics be able to repair path objects ... while still charging for the replacement (due to parts costs) and also spending a little bit of time to do each one (like 2x the time it takes for them to sweep a path). I'd say it makes more in-game sense to have handymen do it ... _but_ handymen already have enough tasks to fill up their task selection window, and handymen don't have an already-existing 'repairing things' animation to use like mechanics do. _Then again_ if mechanics are doing it, you could run into issues where a broken-down ride calls a _much_ farther away mechanic because the closest one was momentarily busy repairing a lamp post.

**Reply by ageekhere** [reply]
Using Security Guards to fix/repair vandalism could be an option.

**Comment by 733737** [comment]
If you've ever played RCT Classic, I think the implementation there is fine

**Comment by Basssiiie** [comment]
There's already a "highlight path issues" option, and then you can repair the benches, lamps and trashbins by placing a new one on top of the broken one.
![image](https://user-images.githubusercontent.com/20048660/181700368-311f6a07-6be5-43ae-ae91-26d7feb476f7.png)

**Comment by 733737** [comment]
I didn't say that they couldn't be repaired, I just like the way RCT Classic handles it better

**Reply by Basssiiie** [reply]
In my screenshot they're also still highlighted, they're just not red.

**Reply by 733737** [reply]
I am not as concerned with the highlighting, i just like clicking to repair

**Comment by ocalhoun6** [comment]
Maybe the scenery window could have a 'repair tool'?

It would be an extra icon along with the other tools (like next to the scatter tool), and if you click that icon, it will change the mouse cursor. Then, if you click on any path tile with any vandalized path item on the tile, it will automatically replace that path scenery item with a new one of the same type (and charge you for it). It could be especially useful in this 'view path issues' mode, but also still very usable in the normal view mode. (Or maybe the main viewport could automatically go into 'view path issues' mode when you turn on the repair tool, then switch back when you deselect the tool?)

Because the main thing that's so tedious about doing it the current way is:
 - If you have different types of objects that are vandalized, you have to select each object in the scenery window, and you might have to do this a _lot_ of times if there's a lot of vandalism affecting different types of objects.
 - You have to be careful where you click or you might accidentally place a new bench/bin/lamp/etc _next to_ the one you're trying to replace, instead of replacing it. Or worse, you might accidentally do something like replacing an intact bench with a new bin.
 
A 'repair tool' solves both of these problems because it automatically selects the right scenery object and because it won't do anything if you click on the wrong tile.

**Comment by karst** [comment]
I personally think it'd be neat if there'd be a checkbox on handymen to automatically repair footpath items. Automatically charges you the replacement cost but would at least not give you a headache keeping your park in check. Like ocal suggested before aswell.
